File: TeachersGuardAPI/Infraestructure/Mappers/AttendanceMapper.cs
using MongoDB.Bson;
using TeachersGuardAPI.App.DTOs.Attendance;
using TeachersGuardAPI.Domain.Entities;
using TeachersGuardAPI.Infraestructure.Models.Mongo;

namespace TeachersGuardAPI.Infraestructure.Mappers
{
    public class AttendanceMapper
    {
        public static AttendanceDocument MapAttendanceEntityToAttendanceDocument(Attendance attendance) => new()
        {
            EntryDate = attendance.EntryDate,
            ExitDate = attendance.ExitDate,
            PlaceId = ObjectId.Parse(attendance.PlaceId),
            UserId = ObjectId.Parse(attendance.UserId),
            IsCompletedAttendance = attendance.FullAttendance
        };

        public static AttendanceDto MapAttendanceEntityToAttendanceDto(Attendance attendance) => new()
        {
            PlaceId = attendance.PlaceId,
            UserId = attendance.UserId,
            AttendanceId = attendance.AttendanceId,
            EntryDate = attendance.EntryDate,
            ExitDate = attendance.ExitDate,
            FullAttendance = attendance.FullAttendance            
        };

        public static Attendance MapAttendanceDocumentToAttendanceEntity(AttendanceDocument attendanceDocument) => new()
        {
            AttendanceId = attendanceDocument.Id.ToString(),
            FullAttendance = attendanceDocument.IsCompletedAttendance,
            PlaceId = attendanceDocument.PlaceId.ToString(),
            UserId = attendanceDocument.UserId.ToString(),
            EntryDate = attendanceDocument.EntryDate,
            ExitDate = attendanceDocument.ExitDate
        };
    }
}

File: lib/infraestructure/mappers/attendance_mapper.dart
import 'package:teachersguard/domain/entities/entities.dart';
import 'package:teachersguard/infraestructure/models/models.dart';

class AttendanceMapper {
  static Attendance mapAttendanceResponseToAttendanceEntity(
          AttendanceObject response) =>
      Attendance(
          id: response.attendanceId,
          entryDate: response.entryDate,
          exitDate: response.exitDate,
          userId: response.userId,
          isAttendanceCompleted: response.fullAttendance,
          place: Place(id: response.placeId, name: response.placeName));

  static WeekAttendance mapWeekAttendanceResponseToWeekAttendanceEntity(
          WeekAttendances response) =>
      WeekAttendance(
          totalAttendances: response.totalAttendances,
          attendances: response.attendances);
}
